public/main.js

```diff
@@ -5,83 +5,66 @@ var todoTitle = document.getElementById("new-todo");
 var error = document.getElementById("error");
 var completeCounter = document.getElementById("count-label");
 var deleteCompletedTodoBtn = document.getElementById("delete-complete-btn");
+var filterButtonAll = document.getElementById("filter-button-all");
+var filterButtonActive = document.getElementById("filter-button-active");
+var filterButtonComplete = document.getElementById("filter-button-completed");
 
 form.onsubmit = function(event) {
     var title = todoTitle.value;
-    createTodo(title, function() {
-        reloadTodoList();
+    var dataToSend = JSON.stringify({
+        title: title
     });
+    makeRequest("api/todo", "POST", dataToSend, "Failed to create item.")
+        .then(reloadTodoList);
     todoTitle.value = "";
     event.preventDefault();
 };
 
 deleteCompletedTodoBtn.onclick = function() {
-    getTodoList(function(todos) {
-        todos.forEach(function(todo) {
-            if (todo.isComplete) {
-                removeTodo(todo.id, reloadTodoList);
-            }
+    makeRequest("api/todo/", "GET", null, "Failed to get list.")
```

Comment:
Just to be clear, this button is deleting all of the completed Todos? 
Is it necessary to call reloadTodoList every delete? it seems like ti would make more sense to do that after all the deletes have been sent.